squidb-processor/src/com/yahoo/squidb/processor/plugins/PluginEnvironment.java

```diff
@@ -212,47 +242,120 @@ private void processPlugin(String pluginName) {
     }
 
     /**
+     * Add a {@link Plugin} class to the list of known plugins
+     *
+     * @param plugin the plugin class
+     * @param priority the priority to give the plugin
+     */
+    public void addPlugin(Class<? extends Plugin> plugin, PluginPriority priority) {
+        switch (priority) {
+            case LOW:
+                lowPriorityPlugins.add(plugin);
+                break;
+            case HIGH:
+                highPriorityPlugins.add(plugin);
+                break;
+            case NORMAL:
+            default:
+                normalPriorityPlugins.add(plugin);
+                break;
+        }
+
+        SupportedOptions supportedOptionsAnnotation = plugin.getAnnotation(SupportedOptions.class);
+        if (supportedOptionsAnnotation != null) {
+            String[] options = supportedOptionsAnnotation.value();
+            if (options != null) {
+                Collections.addAll(pluginSupportedOptions, options);
+            }
+        }
+    }
+
+    private void reportUnsupportedOptions() {
+        Set<String> unsupportedOptions = new HashSet<>(squidbOptions);
+        unsupportedOptions.removeAll(SQUIDB_SUPPORTED_OPTIONS);
+
+        if (AptUtils.isEmpty(unsupportedOptions)) {
+            return;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (String option : unsupportedOptions) {
+            if (sb.length() > 0) {
+                sb.append(SEPARATOR);
+            }
+            sb.append(option);
+        }
+        String message = String.format(UNSUPPORTED_OPTIONS_WARNING, sb);
+        utils.getMessager().printMessage(Diagnostic.Kind.WARNING, message);
+    }
+
+    /**
+     * Returns whether the value of the "squidbOptions" environment option contains the given string
+     *
      * @param option the option to check
      * @return true if the option is set, false otherwise
+     * @deprecated use {@link #hasSquidbOption} instead
      */
+    @Deprecated
     public boolean hasOption(String option) {
```

Comment:
Can we update the various pieces of squidb code that call this deprecated method to use the new name?